libraries/botbuilder-dialogs/src/skillDialog.ts

```diff
@@ -0,0 +1,135 @@
+        const dialogArgs = SkillDialog.validateBeginDialogOptions(options);
+
+        await dc.context.sendTraceActivity(`${ this.id }.beginDialog()`, undefined, undefined, `Using activity of type: ${ dialogArgs.activity.type }`);
+
+        // Store Skill information for this dialog instance
+        await this._activeSkillProperty.set(dc.context, dialogArgs.skill);
+
+        // Create deep clone of the original activity to avoid altering it before forwarding it.
+        const clonedActivity = this.cloneActivity(dialogArgs.activity);
+
+        // Apply conversation reference and common properties from incoming activity before sending.
+        const skillActivity = TurnContext.applyConversationReference(clonedActivity, TurnContext.getConversationReference(dc.context.activity), true) as Activity;
+
+        // Send the activity to the skill.
+        await this.sendToSkill(dc, skillActivity, dialogArgs.skill);
+        return Dialog.EndOfTurn;
+    }
+
+    public async continueDialog(dc: DialogContext): Promise<DialogTurnResult> {
+        await dc.context.sendTraceActivity(`${ this.id }.continueDialog()`, undefined, undefined, `ActivityType: ${ dc.context.activity.type }`);
+
+        // Retrieve the current skill information from ConversationState
+        var skillInfo = await this._activeSkillProperty.get(dc.context, null);
+
+        // Handle EndOfConversation from the skill (this will be sent to the this dialog by the SkillHandler if received from the Skill)
+        if (dc.context.activity.type === ActivityTypes.EndOfConversation) {
+            await dc.context.sendTraceActivity(`${ this.id }.continueDialog()`, undefined, undefined, `Got ${ ActivityTypes.EndOfConversation }`);
+            return await dc.endDialog(dc.context.activity.value);
+        }
+
+        // Forward only Message and Event activities to the skill
+        if (dc.context.activity.type === ActivityTypes.Message || dc.context.activity.type === ActivityTypes.Event) {
+            // Just forward to the remote skill
+            await this.sendToSkill(dc, dc.context.activity, skillInfo);
+        }
+
+        return Dialog.EndOfTurn;
+    }
+
+    /**
+     * Clones the Activity entity.
+     * @param activity Activity to clone.
+     */
+    private cloneActivity(activity: Partial<Activity>): Activity {
+        return Object.assign({} as Activity, activity);
+    }
+
+    private static validateBeginDialogOptions(options: any): SkillDialogArgs {
+        if (!options) {
+            throw new TypeError('Missing options parameter');
+        }
+
+        const dialogArgs = options as SkillDialogArgs;
+
+        if (dialogArgs.skill == undefined || dialogArgs.skill == null) {
+            throw new TypeError(`"skill" undefined or null in options`);
+        }
+
+        if (dialogArgs.activity === undefined || dialogArgs.activity === null) {
+            throw new TypeError(`"activity" is undefined or null in options.`);
+        }
```

Comment:
I'll be changing this to follow the normal falsy checks we do throughout the repository[...](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing)